pkg/resources/stream_on_external_table_acceptance_test.go

```diff
@@ -470,8 +539,92 @@ func TestAcc_StreamOnExternalTable_RecreateWhenStale(t *testing.T) {
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "show_output.0.stale", "true")),
+					assert.Check(resource.TestCheckResourceAttrWith(resourceName, "show_output.0.created_on", func(value string) error {
+						if value == createdOn {
+							return fmt.Errorf("stream was not recreated")
+						}
+						return nil
+					})),
+				),
+			},
+		},
+	})
+}
+
+func TestAcc_StreamOnExternalTable_RecreateWhenStaleWithExternalChanges(t *testing.T) {
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableAcceptance)
+	acc.TestAccPreCheck(t)
+	resourceName := "snowflake_stream_on_external_table.test"
+
+	schema, cleanupSchema := acc.TestClient().Schema.CreateSchemaWithOpts(t,
+		acc.TestClient().Ids.RandomDatabaseObjectIdentifierInDatabase(acc.TestClient().Ids.DatabaseId()),
+		&sdk.CreateSchemaOptions{
+			DataRetentionTimeInDays:    sdk.Pointer(1),
+			MaxDataExtensionTimeInDays: sdk.Pointer(1),
+		},
+	)
+	t.Cleanup(cleanupSchema)
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifierInSchema(schema.ID())
+
+	stageID := acc.TestClient().Ids.RandomSchemaObjectIdentifierInSchema(schema.ID())
+	stageLocation := fmt.Sprintf("@%s", stageID.FullyQualifiedName())
+	_, stageCleanup := acc.TestClient().Stage.CreateStageWithURL(t, stageID)
+	t.Cleanup(stageCleanup)
+
+	externalTable, externalTableCleanup := acc.TestClient().ExternalTable.CreateInSchemaWithLocation(t, stageLocation, schema.ID())
+	t.Cleanup(externalTableCleanup)
+
+	var createdOn string
+
+	model := model.StreamOnExternalTable("test", id.DatabaseName(), externalTable.ID().FullyQualifiedName(), id.Name(), id.SchemaName()).WithInsertOnly(r.BooleanTrue)
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.StreamOnExternalTable),
+		Steps: []resource.TestStep{
+			// initial creation does not lead to stale stream
 			{
+				Config: config.FromModel(t, model),
+				Check: assert.AssertThat(t, resourceassert.StreamOnExternalTableResource(t, resourceName).
+					HasNameString(id.Name()).
+					HasStaleString(r.BooleanFalse),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "show_output.0.stale", "false")),
+					assert.Check(resource.TestCheckResourceAttrWith(resourceName, "show_output.0.created_on", func(value string) error {
+						createdOn = value
+						return nil
+					})),
+				),
+			},
+			// changing the value externally on schema and checking manually that stream is stale
+			{
+				PreConfig: func() {
+					acc.TestClient().Schema.Alter(t, schema.ID(), &sdk.AlterSchemaOptions{
+						Set: &sdk.SchemaSet{
+							DataRetentionTimeInDays:    sdk.Int(0),
+							MaxDataExtensionTimeInDays: sdk.Int(0),
+						},
+					})
+					assert.AssertThatObject(t, objectassert.Stream(t, id).
+						HasName(id.Name()).
+						HasStale(true),
```

Comment:
like we discussed, here it would be better to revert the retention values to non-zero values in the preconfig

Reply:
Done - also for stream on table.